ripd/ripd.c

```diff
@@ -920,9 +920,14 @@ static int rip_auth_md5(struct rip_packet *packet, struct sockaddr_in *from,
 		if (key == NULL || key->string == NULL)
 			return 0;
 
-		strlcpy(auth_str, key->string, sizeof(auth_str));
+		/*
+		 * strncpy is intentional. auth_str should be NULL-terminated
+		 * only when key->string length is less than sizeof(auth_str).
+		 */
```

Comment:
If this is a fixed size buffer why does it need to be null terminated at all?

Reply:
We don't really care about the NULL-byte, but we have to somehow cover the case when the source string is less than 16 bytes. We could manually check the source string length and do something like:
```
memcpy(auth_str, key->string, strlen(key->string) > 16 ? 16 : strlen(key->string));
```
But `strncpy` already does what we need so why not use it?

Reply:
Replaced with `memcpy` as `strncpy` doesn't pass the CI.